src/main/java/seedu/todo/logic/TodoDispatcher.java

```diff
@@ -0,0 +1,28 @@
+package seedu.todo.logic;
+
+import seedu.todo.commons.exceptions.IllegalValueException;
+import seedu.todo.logic.commands.AddTodoCommand;
+import seedu.todo.logic.commands.BaseCommand;
+import seedu.todo.logic.parser.ParseResult;
+
+/**
+ * Selects the correct command based on the parser results
+ */
+public class TodoDispatcher implements Dispatcher {
+    public BaseCommand dispatch(ParseResult parser) throws IllegalValueException {
+        BaseCommand command;
+        
+        switch (parser.getComand()) {
+        case "add":
```

Comment:
I think this is actually correct, if https://github.com/oss-generic/process/issues/3 is anything to go by